equinox/nn/_attention.py

```diff
@@ -325,6 +401,42 @@ def __call__(
                 raise ValueError(
                     "process_heads must not change the shape of the heads."
                 )
+        state_length = None
+        if state is None:
+            causal_mask_offset = 0
+        else:
+            if self.kv_cache is None:
+                raise ValueError(
+                    "State was provided, but cannot use autoregressive decoding "
+                    "without specifying "
+                    "`MultiheadAttention(..., kv_cache=...)`. "
+                    "See `equinox.nn.StandardKVCache` for an example."
+                )
+
+            key_state, value_state, index, state = self.kv_cache(
+                state, key_heads, value_heads
+            )
```

Comment:
Nit: we have `state` as a first argument but as a last output.

The general convention has been to have it at the end -- could this be `self.kv_cache(key_heads, value_heads, state)` instead?